zilencer/views.py

```diff
@@ -475,6 +487,102 @@ def register_remote_push_device(
     return json_success(request)
 
 
+def do_register_remote_push_device(
+    bouncer_public_key: str,
+    encrypted_push_registration: str,
+    push_account_id: int,
+    *,
+    realm: Realm | None = None,
+    remote_realm: RemoteRealm | None = None,
+) -> int:
+    assert settings.PUSH_REGISTRATION_ENCRYPTION_KEYS
+    if bouncer_public_key not in settings.PUSH_REGISTRATION_ENCRYPTION_KEYS:
+        raise MissingBouncerPublicKeyError
+
+    # Decrypt push_registration
+    bouncer_private_key: str = settings.PUSH_REGISTRATION_ENCRYPTION_KEYS[bouncer_public_key]
+    private_key = PrivateKey(bouncer_private_key.encode("utf-8"), encoder=Base64Encoder)
+    unseal_box = SealedBox(private_key)
+
+    try:
+        push_registration_bytes = unseal_box.decrypt(
+            Base64Encoder.decode(encrypted_push_registration.encode("utf-8"))
+        )
+        push_registration = orjson.loads(push_registration_bytes.decode("utf-8"))
+
+        token = push_registration["token"]
+        token_kind = push_registration["token_kind"]
+        validate_token(token, token_kind)
+        # TODO: Add validation for ios_app_id and timestamp
+        ios_app_id = (
+            push_registration["ios_app_id"]
+            if token_kind == RemotePushDevice.TokenKind.APNS
+            else None
+        )
+        timestamp = push_registration["timestamp"]
+    except (TypeError, CryptoError, orjson.JSONDecodeError, KeyError, JsonableError):
```

Comment:
It's a better pattern to have a series of smaller try/except blocks that each only catch the expected exception types.

Reply:
Made the changes to use smaller try/except blocks.